hw_diag/utilities/gcs_shipper.py

```diff
@@ -0,0 +1,48 @@
+import json
+import os
+import datetime
+import requests
+import logging
+from hashlib import sha256
+
+
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
+
+
+BUCKET_NAME = os.getenv(
+    'DIAGNOSTICS_GCS_BUCKET',
+    'inbound-hotspot-diagnostics'
+)
+URL = 'https://www.googleapis.com/upload/storage/v1/b/%s/o?uploadType=media' \
+    % BUCKET_NAME
+
+
+def generate_hash(public_key):
+    hashable = '%s-%s' % (public_key, datetime.datetime.utcnow().isoformat())
+    return sha256(hashable.encode('utf-8')).hexdigest()
+
+
+def upload_diagnostics(diagnostics):
+    log.info('Submitting diagnostics to GCS bucket.')
+    file_name = generate_hash(diagnostics.get('PK'))
```

Comment:
Could you add a comment explaining why we need to hash. Not immediately clear to me what we are trying to accomplish.